### ISSUE TITLE
bridge to rabbitmq, which loses every second message and is  always subscribed as consumer?

### ISSUE THREAD

**Issue by DirkTheDaring** [eclipse/mosquitto#408]
This report  is for the version 1.4.11, which i build  by myself on fedora 25. Addtionally there is a rabbitmq in another container (docker)  which uses the mqtt plugin of rabbitmq to offer a compatible interface.

I have setup a bridge to rabbit mq, which basically has the intention to send all messages  from one topic of mosquitto to rabbitmq. The first strange thing, i noticed was, that no mather if i set "both", "in" or "out" in topic. I see that mosquitto always subscribes as consumer to the queue on rabbitmq  in the rabbitmq console.
The more annoying thing I noticed is, that this actually has an effect, I only receive every second message on the rabbitmq side with my client which subscribes as rabbitmq consumer to the queue.
I believe is due to the reason, that rabbitmq distributes the incoming messages evenly to all  consumers in a  round-robin fashion.   Therefore one message goes to my own consumer, and the other goes back to mosquitto! Not sure about this though, but actually it seems to explain the "lost" messages. 
Actually the topic is created by mosquitto in rabbitmq, so do i miss a special setup parameter?
Or is this a bug, a misunderstanding or how do i turn off this "consumer subscription"  and have  a  "write only" topic bridge to rabbitmq?

Addendum:
I have verified, that is evenly distributed among consumers, by adding more subscriptions to the queue on rabbitmq side. So if add +1 , i get every third message, +2 , every fourth message, ... 

So basically it means, how do i get rid of this "mosquitto subscription"  as consumer?

This is my mosquitto configuration
```
listener 1883
protocol mqtt

listener 8080
protocol websockets
password_file /etc/mosquitto/passwd

log_type debug

connection bridge-to-rabbitmq
address rabbitmq.prod.cloud:1883
username xxx
password xxx
try_private false

topic qaxr out
```

**Comment by ralight** [comment]
I tried this out with 1.4.11 using the configuration below:

```
listener 1889
protocol mqtt

listener 8085
protocol websockets

log_type debug

connection bridge-to-rabbitmq
address localhost:1888
try_private false
topic qaxr out
```

The other end of the bridge was another mosquitto broker. The log from that broker when the bridge connected was this:

```
1490049894: mosquitto version 1.4.11 (build date 2017-03-20 22:33:07+0000) starting
1490049894: Using default config.
1490049894: Opening ipv4 listen socket on port 1888.
1490049894: Opening ipv6 listen socket on port 1888.
1490049975: New connection from 127.0.0.1 on port 1888.
1490049975: New client connected from 127.0.0.1 as ks.ral.me.bridge-to-rabbitmq (c0, k60).
1490049975: Sending CONNACK to ks.ral.me.bridge-to-rabbitmq (0, 0)
1490049975: Received PUBLISH from ks.ral.me.bridge-to-rabbitmq (d0, q1, r1, m1, '$SYS/broker/connection/ks.ral.me.bridge-to-rabbitmq/state', ... (1 bytes))
1490049975: Sending PUBACK to ks.ral.me.bridge-to-rabbitmq (Mid: 1)
1490049975: Received UNSUBSCRIBE from ks.ral.me.bridge-to-rabbitmq
1490049975:     qaxr
1490049975: ks.ral.me.bridge-to-rabbitmq qaxr
```

That says to me that the broker doing the bridging does not connect as a consumer/subscriber to the remote bridge.

Could you provide some log information from the rabbit broker (or ideally from the bridging mosquitto broker as well using the "-v" option) so show that the mosquitto bridge is connecting with a SUBSCRIBE command?

**Comment by DirkTheDaring** [comment]
![image](https://cloud.githubusercontent.com/assets/8758376/24134252/5b1d5402-0e03-11e7-8570-45247f100a45.png)

This is a screen shot of rabbitmq console, with the "Consumers" at the bottom. This consumer shows up as soon, as I run rabbitmq. 
When i add my own consumer,  a second line at the bottom of the screen shot shows up.

log from mosquitto 
```
[root@mosquitto mosquitto]# /usr/sbin/mosquitto -v -c /etc/mosquitto/mosquitto.conf
1490075935: mosquitto version 1.4.11 (build date 2017-03-19 12:49:54+0000) starting
1490075935: Config loaded from /etc/mosquitto/mosquitto.conf.
1490075935: Opening ipv4 listen socket on port 1883.
1490075935: Opening ipv6 listen socket on port 1883.
1490075935: Opening websockets listen socket on port 8080.
1490075935: Bridge local.mosquitto.bridge-to-rabbitmq doing local SUBSCRIBE on topic local/yy/qaxr
1490075935: Connecting bridge bridge-to-rabbitmq (rabbitmq.prod.cloud:1883)
1490075935: Bridge mosquitto.bridge-to-rabbitmq sending CONNECT
1490075935: Received CONNACK on connection local.mosquitto.bridge-to-rabbitmq.
1490075935: Bridge local.mosquitto.bridge-to-rabbitmq sending UNSUBSCRIBE (Mid: 2, Topic: remote/zz/qaxr)
1490075935: Received UNSUBACK from local.mosquitto.bridge-to-rabbitmq
1490075935: Received PUBACK from local.mosquitto.bridge-to-rabbitmq (Mid: 1)
```

**Comment by karlp** [comment]
fwiw, I'm bridging to rabbitmq with no problems at all.  I just had to use "try_private false".

I think the "consumers" item is misleading in the rabbit console, and is purely an artifact of how the mqtt plugin in rabbit works.

**Comment by ralight** [comment]
The log shows that mosquitto doesn't subscribe to anything, so it can't consume anything. I would try `try_private false` as @karlp suggests, or `bridge_attempt_unsubscribe false` in case the UNSUBSCRIBE attempt is confusing rabbitmq.

**Comment by DirkTheDaring** [comment]
by accident i found a fix to the problem, whhen I changed the configuration on the rabbitmq side. I changed the binding to the queue "mqtt-subscription-mosquitto.bridge-to-rabbitmqqos1" to  another name (!= original_name) .  But  I kept to listen to the same routing_key=qaxr . This way it started to work! 
It leaves me confused what is actually happening there though.

**Comment by ralight** [comment]
As you've resolved this I'm closing this issue.
